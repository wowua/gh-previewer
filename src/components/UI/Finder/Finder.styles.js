import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  justify-content: center;
`;

export const Form = styled.form`
  display: inline-flex;  
`;

export const Input = styled.input`
  padding: 6px 12px;
  margin-right: 8px;
  border: 1px solid ${(props) => props.theme.borders};
  border-radius: 5px;
`;

export const Button = styled.button`
  border: 1px solid ${(props) => props.theme.borders};
  text-transform: uppercase;
  padding: 0 10px;
  color: #fff;
  background: ${(props) => props.theme.blue};
  border-radius: 5px;
`;
