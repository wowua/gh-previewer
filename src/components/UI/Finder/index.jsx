import React, { useState } from "react";
import PropTypes from "prop-types";
import { Container, Form, Input, Button } from "./Finder.styles";

const Finder = ({ onSubmit }) => {
  const [title, setTitle] = useState("");

  const submitHandler = e => {
    e.preventDefault();
    const trimmedTitle = title.trim();
    if (trimmedTitle) {
      onSubmit(trimmedTitle);
    }
  }

  return (
    <Container>
      <Form onSubmit={submitHandler}>
        <Input data-cy-id="finder-input" value={title} onChange={e => setTitle(e.target.value)} type="text" placeholder="Enter the name" />
        <Button data-cy-id="finder-btn">Send!</Button>
      </Form>
    </Container>
  )
}

Finder.propTypes = {
  onSubmit: PropTypes.func
}

export default Finder;
