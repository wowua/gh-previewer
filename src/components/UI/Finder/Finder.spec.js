import Finder from "./index.jsx";
import { shallow } from 'enzyme';
import React from "react";

it("finder matches snapshot", () => {
  const wrapper = shallow(<Finder />);
  expect(wrapper).toMatchSnapshot();
});