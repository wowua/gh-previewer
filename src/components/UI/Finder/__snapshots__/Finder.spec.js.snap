// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finder matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.form
      onSubmit={[Function]}
    >
      <styled.input
        data-cy-id="finder-input"
        onChange={[Function]}
        placeholder="Enter the name"
        type="text"
        value=""
      />
      <styled.button
        data-cy-id="finder-btn"
      >
        Send!
      </styled.button>
    </styled.form>
  </styled.div>
</styled.div>
`;
